### PR TITLE
Add citation to DLA-Future to CITATIONS.cff

### PR DESCRIPTION
**Pull request by RMeli**
Also shuffled around some fields.

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -42,5 +44,45 @@ keywords:
...
+        family-names: Nikolov
+      - given-names: Harmen
+        family-names: Stoppels
+      - given-names: Lara
+        family-names: Querciagrossa
+      - given-names: Alo
+        family-names: Roosing
+      - given-names: Alessandro
+        family-names: Colombo
+      orcid: 'https://orcid.org/0000-0002-7950-293X'
```
[the remark this answers is not in the document]
Comment (RMeli) [reply CITATION.cff:88]: Good idea, it will avoid going out-of-sync eventually.

[the remark this answers is not in the document]
Comment (RMeli) [reply CITATION.cff:88]: The authors field seems to be one of the few that is mandatory: https://github.com/citation-file-format/citation-file-format/blob/main/schema-guide.md#definitionsreferenceauthors

[the remark this answers is not in the document]
Comment (RMeli) [reply CITATION.cff:88]: We could go for `name: "DLA-Future Developers"` or something on those lines.

Comment (RMeli) [inline CITATION.cff:88]:
```suggestion
      - name: "DLA-Future Developers"
        affiliation: "CSCS / ETH Zürich"
```

[the remark this answers is not in the document]
Comment (RMeli) [reply CITATION.cff]: I went for the formed and added CSCS/ETHZ as `affiliation`.

**Review by msimberg — APPROVED** [review]

**Pull request merged**